borsh-derive-internal/src/enum_de.rs

```diff
@@ -85,6 +96,11 @@ pub fn enum_de(input: &ItemEnum, cratename: Ident) -> syn::Result<TokenStream2>
             }
             Fields::Unit => {}
         }
+        let discriminant = if use_discriminant {
+            quote! { #discriminant }
+        } else {
+            quote! { #variant_idx }
+        };
         variant_arms.extend(quote! {
             if variant_tag == #discriminant { #name::#variant_ident #variant_header } else
         });
```

Comment:
add following 2 tests to `tests` submodule in this file:
```rust
    #[test]
    fn borsh_discriminant_false() {
        let item_enum: ItemEnum = syn::parse2(quote! {
           #[borsh(use_discriminant = false)]
            enum X {
                A,
                B = 20,
                C,
                D,
                E = 10,
                F,
            }
        })
        .unwrap();
        let actual = enum_de(
            &item_enum,
            Ident::new("borsh", Span::call_site()),
        )
        .unwrap();

        insta::assert_snapshot!(pretty_print_syn_str(&actual).unwrap());
    }
    #[test]
    fn borsh_discriminant_true() {
        let item_enum: ItemEnum = syn::parse2(quote! {
            #[borsh(use_discriminant = true)]
            enum X {
                A,
                B = 20,
                C,
                D,
                E = 10,
                F,
            }
        })
        .unwrap();
        let actual = enum_de(
            &item_enum,
            Ident::new("borsh", Span::call_site())
        )
        .unwrap();

        insta::assert_snapshot!(pretty_print_syn_str(&actual).unwrap());
    }
  ```
  it's solely for the purpose of quick comparison of expansions of two cases.